recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,143 @@
+            raise ConanInvalidConfiguration("Visual Studio build for any MT runtime is not supported")
+        if self.options.with_cjson: # see _configure_cmake for the reason
+            raise ConanInvalidConfiguration("Option with_cjson not yet supported")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["WITH_THREADING"] = self.settings.os != "Windows"
+        self._cmake.definitions["WITH_PIC"] = self.settings.os != "Windows"
+        self._cmake.definitions["WITH_STATIC_LIBRARIES"] = not self.options.shared
+        self._cmake.definitions["WITH_TLS"] = self.options.with_tls
+        self._cmake.definitions["WITH_CLIENTS"] = self.options.clients
+        self._cmake.definitions["WITH_BROKER"] = self.options.broker
+        self._cmake.definitions["WITH_APPS"] = self.options.apps
+        self._cmake.definitions["WITH_PLUGINS"] = self.options.plugins
+        self._cmake.definitions["DOCUMENTATION"] = False
+        if self.options.with_tls:
+            self._cmake.definitions["OPENSSL_SEARCH_PATH"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+            self._cmake.definitions["OPENSSL_ROOT_DIR"] = self.deps_cpp_info["openssl"].rootpath.replace("\\", "/")
+
+        # this does not work, and I do not know how to make cmake of mosquitto FIND_PACKAGE(cJSON) succeed
+        # if self.options.with_cjson:
+        #     self._cmake.definitions["cJSON_SEARCH_PATH"] = self.deps_cpp_info["cjson"].rootpath.replace("\\", "/")
+        #     self._cmake.definitions["cJSON_ROOT_DIR"] = self.deps_cpp_info["cjson"].rootpath.replace("\\", "/")
+
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def _patch(self):
+        if self.settings.os == "Windows":
+            if self.options.with_tls:
+                tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                    "${OPENSSL_LIBRARIES}",
+                                    "${OPENSSL_LIBRARIES} crypt32")
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                    "${OPENSSL_LIBRARIES}",
+                                    "${OPENSSL_LIBRARIES} crypt32")
+                # This is so inconsequent, there is ws2_32 meanwhile in the build, but forgotten here
+                tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_passwd" ,"CMakeLists.txt"),
+                                    "${OPENSSL_LIBRARIES}",
+                                    "${OPENSSL_LIBRARIES} ws2_32 crypt32")
```

Comment:
The `cmake_find_package` does only create `_LIBRARIES` variables with the same casing the module name.
The cmake_find_package generate is missing a lot of features imho.

```suggestion
        if self.options.with_tls:
            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
                                "${OpenSSL_LIBRARIES}",
                                "${OpenSSL_LIBRARIES}")
            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
                                "${OpenSSL_LIBRARIES}",
                                "${OpenSSL_LIBRARIES}")
            tools.replace_in_file(os.path.join(self._source_subfolder, "apps", "mosquitto_passwd" ,"CMakeLists.txt"),
                                "${OpenSSL_LIBRARIES}",
                                "${OpenSSL_LIBRARIES}")
```

Reply:
no, this change does not build the package, I just tested it, costed me quite some time. You dit not test that, right ?

Reply:
No, and I don't have to.